packages/smart-contracts/src/lib/ContractArtifact.ts

```diff
@@ -86,19 +99,36 @@ export class ContractArtifact<TContract extends Contract> {
 
   /**
    * Retrieve the deployment information from the artifact of the used version
-   * deployed into the specified network
+   * deployed into the specified network. Will trow an error if the version of network is incorrect.
    * @param networkName the name of the network where the contract is deployed
-   * @returns the deployment information of the contract as a json object containing address and the number of the creation block
+   * @returns The address and the number of the creation block
    */
   getDeploymentInformation(
     networkName: string,
     version = this.lastVersion,
   ): { address: string; creationBlockNumber: number } {
-    const info = this.info[version].deployment[networkName];
+    const versionInfo = this.info[version];
+    if (!versionInfo) {
+      throw Error(`No deployment for version: ${version}.`);
+    }
+    const info = versionInfo.deployment[networkName];
     // Check the artifact has been deployed into the specified network
     if (!info) {
-      throw Error(`No deployment for network: ${networkName}`);
+      throw Error(`No deployment for network: ${networkName}.`);
     }
     return info;
   }
+
+  /**
+   * Retrieve the deployment information from the artifact of the used version
+   * deployed into the specified network
```

Comment:
Add a small sentence to explain that this one return null instead of throw.

Reply:
it's right below in `returns`